TokenCreation.sol

```diff
@@ -126,7 +126,7 @@ contract TokenCreation is TokenCreationInterface, Token {
                 extraBalance.payOut(address(this), extraBalance.accumulatedInput());
 
             // Execute refund
-            if (msg.sender.send(weiGiven[msg.sender])) { 
+            if (msg.sender.call.value(weiGiven[msg.sender])()) { 
```

Comment:
I have updated. And Fix Transaction-Ordering Dependence, now it can avoid the reentry attack.